src/header.jl

```diff
@@ -383,6 +386,37 @@ function setindex!(hdr::FITSHeader, value::Any, i::Integer)
     hdr.values[i] = value
 end
 
+
+"""
+    delete!(hdr::FITSHeader, key::String)
+
+Delete a key in a FITS header. 
+"""
+function delete!(hdr::FITSHeader, key::String)
+    if (haskey(hdr, key))
+        index = hdr.map[key]
+
+        # Delete the entries
+        deleteat!(hdr.values, index)
+        deleteat!(hdr.keys, index)
+        deleteat!(hdr.comments, index)
+
+        # Then re-create the map array
+        map = Dict{String, Int}()
+        for i in 1:length(hdr.keys)
+            map[hdr.keys[i]] = i
+        end
```

Comment:
Could this work as
```suggestion
        map = Dict(zip(hdr.keys, 1:length(hdr.keys)))
```

Reply:
Yes, that certainly works fine and given the simplicity of the code that is probably tidier.